lib/cachex/services/courier.ex

```diff
@@ -78,6 +89,7 @@ defmodule Cachex.Services.Courier do
               spawn_link(fn ->
                 result =
                   try do
+                    Process.put(:"$callers", callers)
```

Comment:
I think this needs to also add `self()` from the context outside `spawn_link`, right?

Reply:
It is already included in the `callers` set in line 43

Reply:
L43 is the main calling process, but here it goes:

```
fetch/4 (user PID) -> Courier.dispatch/3 (user PID) -> Courier.handle_call/3 (new PID) -> spawn_link (another new PID)
```

We probably want to include that middle one, right?

Reply:
Ah, I see what you mean.

We actually don't care. Since the idea behind `$callers` is to establish a hierarchy between process that defines the mock (somewhere in the root) and process that will use it (somewhere at the leaf), it is not strictly necessary to include ALL processes.

The `Courier` process never runs user provided code directly, so it is not strictly necessary to include it (doesn't hurt either).

BTW, `Task` module propagates `$callers` automatically. So, if we swapped `spawn_link` for `Task.async`, we would have to set `$callers` in the `Courier`.